src/Core/AdminConsole/OrganizationAuth/UpdateOrganizationAuthRequestCommand.cs

```diff
@@ -51,5 +75,65 @@ public async Task UpdateAsync(Guid requestId, Guid userId, bool requestApproved,
                 updatedAuthRequest.RequestIpAddress, deviceTypeAndIdentifier);
         }
     }
-}
 
+    public async Task UpdateAsync(Guid organizationId, IEnumerable<OrganizationAuthRequestUpdate> authRequestUpdates)
+    {
+        var authRequestEntities = await FetchManyOrganizationAuthRequestsFromTheDatabase(organizationId, authRequestUpdates.Select(aru => aru.Id));
+        var processor = new BatchAuthRequestUpdateProcessor(
+            authRequestEntities,
+            authRequestUpdates,
+            new AuthRequestUpdateProcessorConfiguration()
+            {
+                OrganizationId = organizationId,
+                AuthRequestExpiresAfter = _globalSettings.PasswordlessAuth.AdminRequestExpiration
+            }
+        );
+        processor.Process((Exception e) => _logger.LogError(e.Message));
+        await processor.Save((IEnumerable<OrganizationAdminAuthRequest> authRequests) => _authRequestRepository.UpdateManyAsync(authRequests));
+        await processor.SendPushNotifications((ar) => _pushNotificationService.PushAuthRequestResponseAsync(ar));
+        await processor.SendNewDeviceEmails(PushTrustedDeviceEmail);
+        await processor.SendEventLogs(SendOrganizationEventLogs);
+    }
+
+    async Task<ICollection<OrganizationAdminAuthRequest>> FetchManyOrganizationAuthRequestsFromTheDatabase(Guid organizationId, IEnumerable<Guid> authRequestIds)
+    {
+        return authRequestIds != null && authRequestIds.Any() ?
+            await _authRequestRepository
+            .GetManyAdminApprovalRequestsByManyIdsAsync(
+                organizationId,
+                authRequestIds
+            ) :
+            new List<OrganizationAdminAuthRequest>();
+    }
+
+    async Task PushTrustedDeviceEmail<T>(T authRequest, string identifier) where T : AuthRequest
+    {
+        var user = await _userRepository.GetByIdAsync(authRequest.UserId);
```

Comment:
Yeah, I'd prefer to leave this for now. I added some post release work to our epic for adding good batch methods for sending these emails and pushing push notifications. I think we should save this for that work and not mix and match singular and plural methods in one flow. It will be simple to change this when we can also send many emails with one call. I can think of a few ways we could get many users before we can send many emails, but they all kinda smell.